arborcat.module

```diff
@@ -1,120 +1,216 @@
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+          curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
+          $result = curl_exec($ch);
+
+          $simpleXML = simplexml_load_string($result);
+          $i=1;
+          $s = $simpleXML->doors->children();
+          if ($s->assign1==0&$s->assign2==0&$s->assign3==0&$s->assign4==0) {
+              $available=false;
+          } elseif ($s->assign1!==0||$s->assign2!==0||$s->assign3!==0||$s->assign4!==0) {
+              $available=true;
+          }
+      } elseif ($locker_server=='mcblockers') {
+          $ch = curl_init();
+          curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('mcb_lockers'));
+          curl_setopt($ch, CURLOPT_HEADER, 0);
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+          curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
+          $result = curl_exec($ch);
+          if ($result != null) {
+              preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
+              if ($matches[1][0]>0) {
+                  $available = true;
+              }
+          }
       }
-        else if($locker_server=='mcblockers'){
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('mcb_lockers'));
-        curl_setopt($ch, CURLOPT_HEADER, 0);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
-        $result = curl_exec($ch);
-        preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
-        if ($matches[1][0]>0) {
-          $available = TRUE;
-        }
+      return $available;
+  }
+
+  /*
+   * returns a list of request pickup locations.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function arborcat_pickupLocations($destLocation = null)
+  {
+      $db = \Drupal::database();
+      $query = $db->select('arborcat_pickup_location', 'apl');
+      $query->fields('apl', ['locationId', 'branchLocationId', 'timePeriod', 'timePeriodStart', 'timePeriodEnd', 'maxLockers', 'locationName', 'locationDescription']);
+        
+      // add in a condition if a location is supplied to filter on
+      if (3 == strlen($destLocation)) {
+          $query->condition('apl.branchLocationId', (int) $destLocation, '=');
       }
-    return $available;
+      $result = $query->execute();
+      
+      $pickupLocationRecords = $result->fetchAll();
+      return $pickupLocationRecords;
+  }
+
+   /*
+   * Checks the availability of a locker fo the specified date and timePeriod.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function lockerAvailableForDateAndTimeSlot($queryDate, $locationObject)
```

Comment:
function names in .module files should always start with the module name, to prevent collision. Not sure if this is an explicit coding standard or just something that I've always seen and done.